### PR TITLE
Fix search issue in dr-policy

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**
Signed-off-by: Timothy Asir <tjeyasin@redhat.com>

### CODE REVIEW

**Comment by TimothyAsirJeyasing** [comment]
Before:
Screenshot from 2023-02-10 19-56-15

After this fix:
in Apply Dr Policy
Screenshot from 2023-02-10 19-49-09

Screenshot from 2023-02-10 19-48-50

**Comment by TimothyAsirJeyasing** [comment]
@SanjalKatiyar, @GowthamShanmugam  Please review

File: packages/mco/components/modals/drpolicy-apps-apply/subscriptions/application-selector.tsx [hunk 1]
```diff
@@ -202,7 +202,11 @@ export const ApplicationSelector: React.FC<ApplicationSelectorProps> = (
     }));
   };
 
-  const onSearch = (searchValue: string) => {
+  const onSearch = (input: any) => {
+    const searchValue =
+      typeof input === 'string'
+        ? input
+        : (input.target as HTMLInputElement)?.value;
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/drpolicy-apps-apply/subscriptions/application-selector.tsx:209]: add comment...

File: packages/mco/components/modals/drpolicy-apps-status/application-status.tsx [hunk 2]
```diff
@@ -225,7 +225,11 @@ export const ApplicationStatus: React.FC<ApplicationStatusPros> = (props) => {
     setFilteredApplications(applications);
   };
 
-  const onSearch = (searchValue: string) => {
+  const onSearch = (input: any) => {
+    const searchValue =
+      typeof input === 'string'
+        ? input
+        : (input.target as HTMLInputElement)?.value;
```
Comment (SanjalKatiyar) [inline packages/mco/components/modals/drpolicy-apps-status/application-status.tsx:232]: add comment...

File: packages/shared/generic/custom-dual-list-selector.tsx [hunk 3]
```diff
@@ -88,7 +88,11 @@ export const CustomDualListSelector = ({
 
   // builds a search input - used in each dual list selector pane
   const buildSearchInput = (isAvailable: boolean) => {
-    const onChange = (value: string) => {
+    const onChange = (input: any) => {
+      const value =
+        typeof input === 'string'
+          ? input
+          : (input.target as HTMLInputElement).value;
```
Comment (SanjalKatiyar) [inline packages/shared/generic/custom-dual-list-selector.tsx:95]: add comment...

Comment (TimothyAsirJeyasing) [reply packages/shared/generic/custom-dual-list-selector.tsx:99]: Done

File: packages/mco/components/modals/drpolicy-apps-status/application-status.tsx (hunk 2 above)
Comment (TimothyAsirJeyasing) [reply packages/mco/components/modals/drpolicy-apps-status/application-status.tsx:236]: Done

File: packages/mco/components/modals/drpolicy-apps-apply/subscriptions/application-selector.tsx (hunk 1 above)
Comment (TimothyAsirJeyasing) [reply packages/mco/components/modals/drpolicy-apps-apply/subscriptions/application-selector.tsx:213]: Done

**Review by GowthamShanmugam — COMMENTED** [review]
LGTM

**Review by GowthamShanmugam — APPROVED** [review]

**Comment by SanjalKatiyar** [comment]
Jira: https://issues.redhat.com/browse/RHSTOR-4364

**Pull request merged**